packages/server-wallet/src/engine/store.ts

```diff
@@ -357,6 +357,11 @@ export class Store {
     const results = await ObjectiveModel.query(this.knex).where({status: 'approved'});
     return results.map(o => o.toObjective());
   }
+  public async updateObjectiveProgressDate(objectiveIds: string[]): Promise<WalletObjective[]> {
+    return this.knex.transaction(async tx =>
+      Promise.all(objectiveIds.map(oId => ObjectiveModel.updateProgressDate(oId, tx)))
```

Comment:
This seems like it would be an issue for "busy wallets" -- we'd fire off many queries that each update a single row.

Referring to [this code](https://github.com/statechannels/statechannels/blob/eventless-engine/packages/server-wallet/src/wallet/wallet.ts#L252-L256), there are `2N + 2` queries required to `jumpStartObjectives` when there are `N` approved objectives. The `N` queries triggered on this line will likely fail for some modest number `N` (eg. `N = 1000`) with knex timeout issues.

It also feels like there's a concurrency issue here: `updateObjectiveProgressDate` may update the objective concurrent with a `pushMessage` grabbing the objective or something. It's hard to think through the implication of that.

With regards to this PR, it's not **too** hard to get it down to `N+1` queries, by replacing
```
    const objectives = await this._engine.getApprovedObjectives();
    const objectiveIds = objectives.map(o => o.objectiveId);
    await this._engine.store.updateObjectiveProgressDate(objectiveIds);
```
with something like
```
    const objectives = await this._engine.getApprovedObjectives(updateProgressData=true);
```
that returns the single `getApprovedObjectives` query and the `N` queries in
```
       Promise.all(objectiveIds.map(oId => ObjectiveModel.updateProgressDate(oId, tx)))
```

I don't think you need to make that change within this PR, because `2N+2` queries and `N+1` queries are just about as bad as one another.